plugins/modules/database/influxdb/influxdb_retention_policy.py

```diff
@@ -140,6 +204,21 @@ def create_retention_policy(module, client):
     default = module.params['default']
     shard_group_duration = module.params['shard_group_duration']
 
+    if not check_duration_literal(duration):
+        module.fail_json(msg="Failed to parse value of duration")
+
+    influxdb_duration_format = parse_duration_literal(duration)
+    if influxdb_duration_format != 0 and influxdb_duration_format < 3600000000000:
+        module.fail_json(msg="duration value must be at least 1h")
```

Comment:
Is "at least 1h" a requirement of the module, or influxdb? Generally we should avoid to check such bounds that are also checked by the program/service we are working with, since they can change for that service. If they ever allow smaller values, we suddenly have a bug.

Reply:
It's an InfluxDB requirement. The reason I included these checks is that InfluxDB behavior is strange regarding this: if `duration` option is set to less than `1h` requests fail with an error but, if an invalid value is set in the `shard_group_duration` no errors are thrown, the request completes successfully and that value is silently changed to `1h`.

This would result result in an always-changing task if someone sets a `shard_group_duration` value of, for example, `15m`.

Also, I think InfluxDB 1.x behavior should be considered stable at this point and this constraint wont't change in the future. InfluxDB 2.0 has been released recently and database/retention policy pairs have been changed to a new concept, `buckets`, so this module can only be used with 1.x versions. 1.x branch is still be supported by InfluxData but only for maintenance releases.